src/App.tsx

```diff
@@ -1,35 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { ThemeProvider, createTheme } from '@mui/material';
 import CssBaseline from '@mui/material/CssBaseline';
 import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom';
-import { theme } from '@pagopa/mui-italia';
+import { LangCode, theme } from '@pagopa/mui-italia';
 import { Layout } from './components/Layout';
 import { useTranslation } from 'react-i18next';
 
 import './translations/i18n';
 import { ArcRoutes } from './routes/routes';
 import Transaction from './routes/Transaction';
+import i18n from './translations/i18n';
 
-const customTheme = createTheme({
-  ...theme,
-  palette: {
-    ...theme.palette,
-    background: {
-      paper: '#F5F5F5', // mui-italia paper is defined as #FFFFFF
-      default: '#FFFFFF' // mui-italia default is defined as '#F2F2F2'
-    }
-  }
-});
+const customTheme = createTheme(theme);
 
 export const App = () => {
   const { t } = useTranslation();
   document.title = t('app.title');
+  const [currentLanguage, setCurrentLanguage] = useState<LangCode>('it');
+
+  const changeLanguage = (langCode: LangCode) => {
+    i18n.changeLanguage(langCode);
+    setCurrentLanguage(langCode);
+  };
```

Comment:
Please move this logic in an external hook so you can write some unit test easily